appinventor/appengine/src/com/google/appinventor/client/youngandroid/TextValidators.java

```diff
@@ -187,15 +188,20 @@ public static String getErrorMessage(String filename){
     String errorMessage = "";
     String noWhitespace = "[\\S]+";
     String firstCharacterLetter = "[A-Za-z].*";
-    if(!filename.matches("[A-Za-z][A-Za-z0-9_]*") && filename.length() > 0) {
-      if(!filename.matches(noWhitespace)) { //Check to make sure that this project does not contain any whitespace
-        errorMessage = MESSAGES.whitespaceProjectNameError();
-      } else if (!filename.matches(firstCharacterLetter)) { //Check to make sure that the first character is a letter
-        errorMessage = MESSAGES.firstCharProjectNameError();
+    String temp = filename.replaceAll("( )+", " ").replace(" ","_");
+    if(temp.length() > 0){
```

Comment:
Make sure to follow the Google Java style guide here w.r.t. spacing around characters and indentation.